base/errorshow.jl

```diff
@@ -208,6 +208,32 @@ function show_convert_error(io::IO, ex::MethodError, @nospecialize(arg_types_par
     end
 end
 
+"""
+    methoderror_hints
+
+Packages can provide hints about appropriate corrective actions for
+specific `MethodError`s with `push!(Base.methoderror_hints, hintmessage)`, where
+`hintmessage` should be a function like
+
+```julia
+function hintmessage(f, arg_types, kwargs)
```

Comment:
How about passing an `IO` object here so that the hint message can use, e.g., colors?  It can be either the `io` which `showerror` is printing to or a separate buffer so that `showerror` can do appropriate post-processing (e.g., add a newline only if needed).  Alternatively, this can be done by passing a `context` object like `sprint`.